src/features/creators/existingProjects.ts

```diff
@@ -22,7 +26,30 @@ export class ExistingProjects implements PythonProjectCreator {
             return;
         }
 
-        return results.map((r) => ({
+        // do we have any limitations that need to be applied here?
+        // like selected folder not child of workspace folder?
+
+        const filtered = results.filter((uri) => {
+            const p = this.pm.get(uri);
+            if (p) {
+                // Skip this project if there's already a project registered with exactly the same path
+                const np = path.normalize(p.uri.fsPath);
+                const nf = path.normalize(uri.fsPath);
+                return np !== nf;
+            }
+            return true;
+        });
+
+        if (filtered.length === 0) {
+            // No new projects found that are not already in the project manager
+            traceInfo('All discovered projects are already registered in the project manager');
+            setImmediate(() => {
+                showWarningMessage('No new projects found');
```

Comment:
I found it confusing as a user to see nothing when no projects were found but not sure exactly how to surface that error to the user